addons/event_booth/models/event_event.py

```diff
@@ -0,0 +1,94 @@
+        if self.ids or self._origin.ids:
+            booth_tokeep_ids = self.env['event.booth.category'].search([
+                ('event_id', 'in', self.ids or self._origin.ids),
+                ('event_booth_ids', '!=', False),
+            ]).ids
+        else:
+            booth_tokeep_ids = []
+        for event in self:
+            if booth_tokeep_ids:
+                booth_toremove_ids = event._origin.event_booth_category_ids.filtered(lambda booth: booth.id not in booth_tokeep_ids)
+                command = [Command.unlink(booth.id) for booth in booth_toremove_ids]
+            else:
+                command = [Command.clear()]
+            if event.event_type_id.use_booth:
+                booth_category_fields = self.env['event.booth.category']._get_event_booth_category_fields_whitelist()
+                command += [
+                    Command.create({
+                        attribute_name: line[attribute_name] if not isinstance(line[attribute_name], models.BaseModel)
+                        else line[attribute_name].id for attribute_name in booth_category_fields
+                    }) for line in event.event_type_id.event_booth_category_ids if line.name not in event.event_booth_category_ids.mapped('name')
+                ]
+            event.event_booth_category_ids = command
+
+    @api.depends('event_booth_ids', 'event_booth_ids.state')
+    def _compute_event_booth_count(self):
+        booths = self.env['event.booth'].sudo().read_group(
+            [('event_id', 'in', self.ids)],
+            ['event_id', 'state'], ['event_id', 'state'], lazy=False
+        )
+        booths_count = dict()
+        for booth in booths:
+            event_id = booth['event_id'][0]
+            booths_count.setdefault(event_id, {'total': 0, 'available': 0})
+            if booth['state'] == 'available':
+                booths_count[event_id].update(available=booth['__count'])
+            booths_count[event_id]['total'] += booth['__count']
+
+        for event in self:
+            booth_count = booths_count.get(event.id)
+            if booth_count:
+                event.event_booth_count_available = booth_count.get('available')
+                event.event_booth_count = booth_count.get('total')
+
+    @api.depends('event_booth_slot_ids', 'event_booth_slot_ids.state')
+    def _compute_event_booth_slots_count(self):
+        booth_slots = self.env['event.booth.slot'].sudo().read_group(
+            [('event_id', 'in', self.ids)],
+            ['event_id', 'state'], ['event_id', 'state'], lazy=False
+        )
+        booths_slot_count = dict()
+        for booth_slot in booth_slots:
+            event_id = booth_slot['event_id'][0]
+            booths_slot_count.setdefault(event_id, {'total': 0, 'available': 0})
+            if booth_slot['state'] == 'available':
+                booths_slot_count[event_id].update(available=booth_slot['__count'])
+            booths_slot_count[event_id]['total'] += booth_slot['__count']
```

Comment:
same here: why override the available instead of increment it ?
also this part could be factorized with the previous method, as, beside the naming, this is exactly the same code.

```
def _get_booth_stat_count(self, model_name):
    elements = self.env[model_name].sudo().read_group(
        [('event_id', 'in', self.ids)],
        ['event_id', 'state'], ['event_id', 'state'], lazy=False
    )
    elements_count = dict()
    for element in elements:
        event_id = element['event_id'][0]
        elements_count.setdefault(event_id, {'total': 0, 'available': 0})
        if element['state'] == 'available':
            elements_count[event_id]['available'] += element['__count'])
        elements_count[event_id]['total'] += element['__count']

def _compute_event_booth_count(self):
    booth_count = self._get_booth_stat_count('event.booth')
    for event in self:
        ...

def _compute_event_booth_slots_count(self):
    booth_slots_count = self._get_booth_stat_count('event.booth.slot')
    for event in self:
        ...
```

Reply:
and _get_booth_stat_count could event be reused in event.booth model if you give the field in param.